reyna/src/com/b2msolutions/reyna/Dispatcher.java

```diff
@@ -52,30 +61,46 @@ protected Result sendMessage(Message message, HttpPost httpPost, HttpClient http
     public static Result canSend(Context context) {
         ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
         NetworkInfo info = connectivityManager.getActiveNetworkInfo();
-        if(info == null || !info.isConnectedOrConnecting()) {
+        if (info == null || !info.isConnectedOrConnecting()) {
             return Result.NOTCONNECTED;
         }
+        int type = info.getType();
+        Preferences preferences = new Preferences(context);
 
-        TimeRange range = new Preferences(context).getCellularDataBlackout();
-        if (range == null) {
+        if (power.isCharging(context) && !preferences.canSendOnCharge()) return Result.BLACKOUT;
+        if (!power.isCharging(context) && !preferences.canSendOffCharge()) return Result.BLACKOUT;
+        if (info.isRoaming() && !preferences.canSendOnRoaming()) return Result.BLACKOUT;
+        try {
+            BlackoutTime blackoutTime = new BlackoutTime();
+            if (type == ConnectivityManager.TYPE_WIFI && !blackoutTime.canSendAtTime(new GregorianCalendar(), preferences.getWlanBlackout())) return Result.BLACKOUT;
+            if (isTypeMobile(type) && !blackoutTime.canSendAtTime(new GregorianCalendar(), preferences.getWwanBlackout())) return Result.BLACKOUT;
+        } catch (ParseException e) {
+            Logger.w(TAG, "canSend", e);
             return Result.OK;
         }
 
-        int type = info.getType();
+        TimeRange range = preferences.getCellularDataBlackout();
```

Comment:
I understand @kbeatrice comment "@youhana.hana: extend the current cellular blackout to match the others. " the way that we don't need two different options for the new WwanBlackout and the old CellularDataBlackout, but it would be enough to have the old config file option accept multiple ranges.